### ISSUE TITLE
[Question] Is there a way to use your object generator to generate objects for an arbitrary OpenAPI Spec?

### ISSUE THREAD

**Issue by matt-carr** [hnaderi/scala-k8s#236]
I'm writing an integration with Argo Workflows and I was thinking it'd be handy to reuse your object generator to process their [OpenAPI spec](https://raw.githubusercontent.com/argoproj/argo-workflows/master/api/openapi-spec/swagger.json) but I wasn't able to find a way to integrate with it (I'm still fairly new to the scala ecosystem so I may just have missed it)

**Comment by hnaderi** [comment]
There are some general object generation tools in the ecosystem, but they are very general and on the other hand, parsing the spec is very easy as it is just normal json and we can just model what we want to create using a few transformations.

 Object generation in this library is specifically made for kubernetes and is not suited for general object generation from OpenAPI, as there are some specific details about kubernetes spec that are not handled in general object generators, like types, helpers and custom patch commands.

However you can use the OpenAPI parsing and sbt plugin section as an idea or even a starting point to build your own.
This is the data model for parsing [OpenAPI](https://github.com/hnaderi/scala-k8s/blob/main/project/generator/src/main/scala/Model.scala), which is gen translated into [kubernetes object models](https://github.com/hnaderi/scala-k8s/blob/main/project/generator/src/main/scala/DataModel.scala) and then several different representation of each is generated (object, pointers, encoders, decoders, ...).
All of these are located in [here](https://github.com/hnaderi/scala-k8s/tree/main/project)

**Comment by matt-carr** [comment]
Thanks for the quick response! Yeah I skimmed through the generation code, it generally made sense but I was struggling to find an entrypoint - I picked up https://github.com/eikek/sbt-openapi-schema but for a giggle I'll see if I can extend your stuff a bit as well

**Comment by matt-carr** [comment]
Spent a whole bunch of time levelling up on sbt and figured it out - not in a nice way or anything, I just ran the object generation plugin, then replaced the downloaded file with the argo spec, reran it, and scooped the generated objects. 

```bash
sbt 'project objectsJVM' '++ 3'
<wait>
<replace api spec under /specifications>
sbt 'project objectsJVM' '++ 3'
cp modules/objects/.jvm/target/scala-2.12/src_managed/main <target>
```

It actually worked really well! Not perfect but pretty darn good, considering - definitely better than any of the other openapi-scala generators I found.
